CMakeLists.txt

```diff
@@ -268,6 +268,11 @@ else()
     endif()
 endif()
 
+# Remove optimization level 3 added by default by cmake
+string(REPLACE "-O3" "" CMAKE_C_FLAGS_RELEASE "${CMAKE_C_FLAGS_RELEASE}")
+# Set optimization level 2 for release mode
+set(CMAKE_C_FLAGS_RELEASE "${CMAKE_C_FLAGS_RELEASE} -O2")
```

Comment:
Should this be made conditional perhaps? Only add -O2 if it does not already have it?
Also, can this still be overridden by command-line?

I seem to remember that there is a way to set the defaults (for when `cmake .` is run the first time), would that be more appropriate perhaps? I am not sure, but I suspect this would also change what is in the cache if run a second time.

Reply:
I could do something like this:
```
if(NOT CMAKE_C_FLAGS MATCHES "([\\/\\-]O)[0-9d]")
    # Remove optimization level 3 added by default by cmake
    string(REGEX REPLACE "([\\/\\-]O)3" "\\2"
        CMAKE_C_FLAGS_RELEASE ${CMAKE_C_FLAGS_RELEASE})
else()
    # Optimization level provided by C flags, remove it from release flags
    string(REGEX REPLACE "([\\/\\-]O)[0-9d]" ""
        CMAKE_C_FLAGS_RELEASE ${CMAKE_C_FLAGS_RELEASE})
endif()
```

Or I can just reset the default release mode C flags:

```
if(MSVC)
    set(CMAKE_C_FLAGS_RELEASE "/O2 /DNDEBUG")
else()
    set(CMAKE_C_FLAGS_RELEASE "-O2 -DNDEBUG")
endif()
```
In this instance anybody can set `-DCMAKE_C_FLAGS=/O3` and this is added after the `CMAKE_C_FLAGS_RELEASE`:

![image](https://user-images.githubusercontent.com/1364432/86503267-817aaf80-bd61-11ea-8eb8-77be0ab72787.png)

Which do you prefer?

Reply:
I have made some changes.